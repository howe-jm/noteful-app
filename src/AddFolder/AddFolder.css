.add-folder-form {
  display: flex;
  flex-flow: column wrap;
  align-content: center;
  color: #cacaca;
  padding-bottom: 30px;
  width: 400px;
  background-color: #010d1e;
  text-align: center;
  margin: 10px 0 0 10px;
}

.error-text {
  font-size: 0.8rem;
  color: black;
  background: lightcoral;
  border-radius: 20px;
  padding: 5px 5px 8px 5px;
}
