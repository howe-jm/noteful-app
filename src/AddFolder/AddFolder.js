import React from 'react';
import './AddFolder.css';
import ApiContext from '../ApiContext';
import cuid from 'cuid';
import ErrorPage from '../ErrorBoundary/ErrorBoundary';
import PropTypes from 'prop-types';

export default class AddFolder extends React.Component {
  constructor(props) {
    super(props);
    this.state = { value: '', error: false, submitting: false, apiError: false, apiErrMsg: '' };
  }

  static contextType = ApiContext;

  handleSubmit = (event) => {
    event.preventDefault();
    this.postFormToApi(this.state.value);
  };
  postFormToApi = (name) => {
    this.setState({ submitting: true });
    const myHeaders = new Headers();
    myHeaders.append('Content-Type', 'application/x-www-form-urlencoded');

    const urlencoded = new URLSearchParams();
    urlencoded.append('id', `${cuid()}`);
    urlencoded.append('name', `${name}`);

    const requestOptions = { method: 'POST', headers: myHeaders, body: urlencoded, redirect: 'follow' };

    fetch('http://localhost:9090/folders/', requestOptions)
      .then((res) => {
        if (!res.ok) return res.json().then((e) => Promise.reject(e));
        return res.json();
      })
      .then((res) => {
        this.context.addFolder(res);
      })
      .then(() => this.props.history.push(`/`))
      .catch((error) => this.setState({ submitting: false, apiError: true, apiErrMsg: `${error}` }));
  };

  formErrorState = (event) => {
    event.preventDefault();
    this.setState({ error: true });
  };

  render() {
    const { submitting, apiError, apiErrMsg } = this.state;
    return (
      <ErrorPage>
        <div className='add-folder-form'>
          <h2 className='add-form-title'>Add Folder</h2>
          <form onSubmit={!this.state.value || this.state.value === '' ? this.formErrorState : this.handleSubmit}>
            <label>
              Folder Name
              <div>
                <input type='text' name='name' onChange={(event) => this.setState({ error: false, value: event.target.value })} />
              </div>
            </label>
            <div>
              <input type='submit' value='Submit' />
            </div>
            {submitting ? <h2 className='loading-text'>Submitting...</h2> : apiError ? <h2 className='error-text'>Error: {apiErrMsg}</h2> : null}
            {this.state.error && <p className='error-text'>Folder must have a name!</p>}
          </form>
        </div>
      </ErrorPage>
    );
  }
}

AddFolder.defaultProps = {
  history: PropTypes.object,
};
