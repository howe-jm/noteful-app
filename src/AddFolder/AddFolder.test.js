import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddFolder from './AddFolder';

describe(`Add Folder component`, () => {
  it('Renders without errors', () => {
    const wrapper = shallow(<AddFolder />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
