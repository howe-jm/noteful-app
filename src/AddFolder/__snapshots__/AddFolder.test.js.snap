// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Folder component Renders without errors 1`] = `
<ErrorPage>
  <div
    className="add-folder-form"
  >
    <h2
      className="add-form-title"
    >
      Add Folder
    </h2>
    <form>
      <label>
        Folder Name
        <div>
          <input
            name="name"
            onChange={[Function]}
            type="text"
          />
        </div>
      </label>
      <div>
        <input
          type="submit"
          value="Submit"
        />
      </div>
    </form>
  </div>
</ErrorPage>
`;
